website/docs/getting_started.html.markdown

```diff
@@ -0,0 +1,231 @@
+automatically spread across all of GCP.
+
+-> Want to try out another location? Check out the [list of available regions and zones](https://cloud.google.com/compute/docs/regions-zones/#available).
+Instances created in zones outside the US are not part of the always free tier
+and could incur charges.
+
+## Creating a VM instance
+A [Google Compute Engine VM instance](https://cloud.google.com/compute/docs/instances/) is
+named `google_compute_instance` in Terraform. The `google` part of the name
+identifies the provider for Terraform, `compute` indicates the GCP product
+family, and `instance` is the resource name.
+
+Google provider resources will generally, although not always, be named after
+the name used in the `gcloud`/the REST API. For example, a VM instance is called
+[`instance` in the API](https://cloud.google.com/compute/docs/reference/rest/v1/instances).
+Most resource field names will also correspond 1:1 with their `gcloud`/REST API
+names.
+
+If you look at the [`google_compute_instance documentation`](/docs/providers/google/r/compute_instance.html),
+you'll see that `project` and `zone` (VM instances are a zonal resource) are
+listed as optional. You can omit them from a Terraform config and the provider
+defaults will be used. If any of these fields is added to a config, it will be
+used instead of the default.
+
+Try adding the following to your config file:
+
+```hcl
+resource "google_compute_instance" "vm_instance" {
+  name         = "terraform-instance"
+  machine_type = "f1-micro"
+
+  boot_disk {
+    initialize_params {
+      image = "debian-cloud/debian-9"
+    }
+  }
+
+  network_interface {
+    # A default network is created for all GCP projects
+    network       = "default"
+    access_config = {
+    }
+  }
+}
+```
+
+~> Note: Don't use `terraform apply` quite yet! You still need to add GCP
+credentials. If you want to try out provisioning your VM instance before
+continuing, follow the instructions in the "Adding credentials" section below.
+
+## Connecting to a VPC network
+
+Like this VM instance, nearly every GCP resource will have a `name` field. They
+are used as a short way to identify resources, and their name in the Cloud
+Console will be the one defined in the `name` field.
+
+When linking resources in a Terraform config though, you'll primarily want to
+use a different field, the `self_link` of a resource. Like `name`, nearly every
+resource has a `self_link`. They look like:
+
+```
+{{API base url}}/projects/{{your project}}/{{location type}}/{{location}}/{{resouce type}}/{{name}}
+```
+
+For a real example, let's look at what the instance `self_link` will be assuming
+your project is named `foo`:
+
+```
+https://www.googleapis.com/compute/v1/projects/foo/zones/us-central1-c/instances/terraform-instance
+```
+
+A resource's `self_link` is a unique reference to that resource. When
+connecting two resources in Terraform, you can use Terraform interpolation to
```

Comment:
I think we definitely want to call out interpolating self links explicitly here since that's what makes Terraform figure out how to make the resources in the right order. Unless I'm misunderstanding your comment?

Reply:
I'm thinking that any reference of resources from another resource (by name, id, self_link, etc) is pretty standard terraform and doesn't need much explaining here. IIRC the DAG building stuff is well documented in a few videos / posts. The fact that we use self_link's referentially instead of id's is the important part and it doesn't matter what the self_link actually is.

Reply:
Oh sure! I guess it doesn't matter too much. I'm going to keep it b/c my personal preference would be see the format. Being told "there are these unique references to resources" and then not seeing their shape would make me immediately look up their shape.

Reply:
I'm just going to leave `id` out because the less references to it the better imo.

Reply:
I was interpreting this as a document for people who have experience in other terraform providers and how do we teach them what is unique to GCP. All the other providers use `id` pretty commonly, so I think an explicit directive to _not_ use `id` is valuable to a user coming from Azure/Aws/etc.

This is all based on my assumption that we are actively discouraging usage of the id field. 

examples from other docs:

```
...
resource "azurerm_network_interface" "myterraformnic" {
    name                      = "myNIC"
    location                  = "eastus"
    resource_group_name       = "${azurerm_resource_group.myterraformgroup.name}"
    network_security_group_id = "${azurerm_network_security_group.myterraformnsg.id}"

    ip_configuration {
        name                          = "myNicConfiguration"
        subnet_id                     = "${azurerm_subnet.myterraformsubnet.id}"
        private_ip_address_allocation = "dynamic"
        public_ip_address_id          = "${azurerm_public_ip.myterraformpublicip.id}"
    }
}
```

[Terraform's getting started doc](https://www.terraform.io/intro/getting-started/dependencies.html):
```
resource "aws_eip" "ip" {
  instance = "${aws_instance.example.id}"
}
```

Reply:
Sure, but some users will also read this as a first time user. They should not see a reference to `id`. I just don't think that we should mention undocumented fields that conveniently exist and also work because they are almost always `self_link`s in an introduction to the provider.

The [eip docs](https://www.terraform.io/docs/providers/aws/r/eip.html#attributes-reference) document an `id` field. And the `instance` field [says it accepts an id](https://www.terraform.io/docs/providers/aws/r/eip.html#attributes-reference). Our resources reference `self_link` and `name`, so the intent of this documentation is to encourage `self_link`s are used over `name` (because if anything they use is ever cross project or region etc users will not be surprised by their provider defaults if they use self links).

Reply:
I don't think we need to bike shed too much further on this. I'm happy to discuss more IRL or if other people have opinions to let majority decide.

I will propose that if we have to keep reminding people _not_ to use id in issues then it would be good to document it somewhere so we have a resource to point people to.